lib/internal/process/promises.js

```diff
@@ -41,88 +40,179 @@ const { isErrorStackTraceLimitWritable } = require('internal/errors');
 // *Must* match Environment::TickInfo::Fields in src/env.h.
 const kHasRejectionToWarn = 1;
 
-const maybeUnhandledPromises = new SafeWeakMap();
-const pendingUnhandledRejections = [];
-const asyncHandledRejections = [];
-let lastPromiseId = 0;
+let hasMultipleResolvesListener = false;
 
-// --unhandled-rejections=none:
-// Emit 'unhandledRejection', but do not emit any warning.
-const kIgnoreUnhandledRejections = 0;
+if (process.on) {
+  hasMultipleResolvesListener = process.listenerCount('multipleResolves') !== 0;
 
-// --unhandled-rejections=warn:
-// Emit 'unhandledRejection', then emit 'UnhandledPromiseRejectionWarning'.
-const kAlwaysWarnUnhandledRejections = 1;
+  process.on('newListener', (eventName) => {
+    switch (eventName) {
+      case 'multipleResolves':
```

Comment:
Why use a `switch` for a single value? Do we expect to add more?

Reply:
in a previous iteration i was also listening to 'unhandledRejection' event. I changed it to a if condition.